src/classes/CallableDispatchService_ERR.cls

```diff
@@ -0,0 +1,207 @@
+     * based on an "ActionType" parameter
+     * @param CallableApiParameters paramService Parameteters passed to the Callable API as part of it's invocation
+     * @return Object The result of the dispatch action's invocation
+     */
+    public static Object dispatch(CallableApiParameters paramService) {
+
+        String dispatchAction = paramService.getString(PARAM_DISPATCH_ACTION_TYPE);
+
+        String cleanDispatchAction = (String.isBlank(dispatchAction) ? '' : dispatchAction.toLowerCase());
+
+        // Pulling Context out of switch since all the dispatch actions need the Context parameter
+        String context = paramService.getString(PARAM_CONTEXT_NAME);
+
+        switch on (cleanDispatchAction) {
+            when 'adderror' { // Add Error
+                return  addError(   context,
+                                    paramService.getString(PARAM_MESSAGE)
+                        );
+
+            } when 'addexception' { // Add Exception
+                return  addException(   context,
+                                        paramService.getException(PARAM_EXCEPTION)
+                        );
+
+            } when 'addrecorderrors' { // Add Record Errors
+                return  addRecordErrors(    context,
+                                            paramService.getSObjects(PARAM_SOBJECTS),
+                                            paramService.getStringsById(PARAM_ERR_MSGS_BY_RECORDID)
+                        );
+
+            } when 'addrecorddmlerrors' { // Add Record Dml Errors
+                return  addRecordDmlErrors( context,
+                                            paramService.getSObjects(PARAM_SOBJECTS),
+                                            paramService.getObjects(PARAM_DML_RESULTS)
+                        );
+
+            } when 'clearrecorderrors' { // Clear Record Errors
+                return  clearRecordErrors(  context,
+                                            paramService.getSObjects(PARAM_SOBJECTS),
+                                            paramService.getString(PARAM_MESSAGE)
+                        );
+
+            } when else { // Invalid Action Type / Action Type Not implemented
+                throw new MalformedErrorDispatchMethodApiException(
+                    String.format(System.Label.CallableApiMethodNotImplemented, new List<String> { 'err.dispatchaction => ' + dispatchAction})
+                );
+            }
+        }
+    }
+
+    /**
+     * @description Adds an Error Message via the ERR_Handler class.
+     *              @see ERR_Handler.processError
+     * @param context The context the exception should be logged under
+     * @param errorMessage The Error to Log
+     * @return Boolean true
+     */
+    private static Boolean addError(String context, String errorMessage) {
+        ERR_Handler.processError(errorMessage, context);
+
+        return true;
```

Comment:
I was leaving myself open to a future scenario where I did need to return an Object back via the Dispatcher...so I thought they all should return something...I could restructure it though.

Reply:
I refactored these to be void, and not return a value.  The dispatch action does, however, return true at the end (similar to the Callable API).  I feel strongly it should, so that we don't have to update the Callable API to go from a method invocation to a return method invocation if needed.